tests/test_text_encoder_lora.py

```diff
@@ -0,0 +1,101 @@
+# coding=utf-8
+# Copyright 2023 HuggingFace Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+import os
+import tempfile
+import unittest
+
+import torch
+from transformers import CLIPTextConfig, CLIPTextModel
+
+from diffusers.loaders import TextEncoderLoRAMixin
+from diffusers.utils import torch_device
+
+
+class TextEncoderLoRATests(unittest.TestCase):
+    def get_dummy_components(self):
+        torch.manual_seed(0)
+        text_encoder_config = CLIPTextConfig(
+            bos_token_id=0,
+            eos_token_id=2,
+            hidden_size=32,
+            intermediate_size=37,
+            layer_norm_eps=1e-05,
+            num_attention_heads=4,
+            num_hidden_layers=5,
+            pad_token_id=1,
+            vocab_size=1000,
+        )
+        text_encoder = CLIPTextModel(text_encoder_config).to(torch_device)
+        text_encoder_lora_wrapper = TextEncoderLoRAMixin(copy.deepcopy(text_encoder))
```

Comment:
I agree with @patrickvonplaten and @patil-suraj, if this is the way to use the new class then it shouldn't be a mixin.